core/core_modules/admin.js

```diff
@@ -0,0 +1,170 @@
+                return true;
+            }
+    }
+},
+
+matchHook = function (moduleName, origionalMatch, config) {
+    return function (event, commandPrefix) {
+        if (getHasPermission(config, event.sender_id, event.sender_name, event.thread_id, moduleName)) {
+            return origionalMatch(event, commandPrefix);
+        }
+        console.log('no permission')
+        return false;
+    }
+},
+
+helpHook = function (moduleName, origionalHelp, config) {
+    return function (commandPrefix) {
+        if (getHasPermission(config, event.sender_id, event.sender_name, event.thread_id, moduleName)) {
+            return origionalHelp.call(this, commandPrefix);
+        }
+        return false;
+    }
+};
+
+exports.load = function () {
+    exports.config = config.getConfig('admin');
+    for (var i = 0; i < exports.platform.coreModules.length; i++) {
+        var help = exports.platform.coreModules[i].help,
+            match = exports.platform.coreModules[i].match;
+
+        exports.platform.coreModules[i].help = helpHook(exports.platform.coreModules[i].name, help, exports.config);
+        exports.platform.coreModules[i].match = matchHook(exports.platform.coreModules[i].name, match, exports.config);
+    }
+};
+
+exports.match = function (event, commandPrefix) {
+    return event.arguments[0] === commandPrefix + 'admin';
+};
+
+exports.help = function () {
+    // Deliberately do not provide details on the usage of this, so users do not know of its existance.
+    return [];
+};
+
+exports.run = function (api, event) {
+    if (event.arguments.length !== 4) {
+        api.sendMessage('Usage:\n' +
+            '- admin <grant/revoke> <fullName/userId> <permissionName>\n' +
+            '- admin <create/delete> <coreModuleName> <permissionName>\n' +
+            'For spaces in any section, encapsulate with double quotes (").', event.thread_id);
+        return false;
+    }
+
+    var action = (event.arguments[1] + '').trim().toLowerCase();
+    var permission = (event.arguments[3] + '').trim().toLowerCase();
+
+    if (action === 'create' || action === 'delete') {
+        if (!setup(action, event.arguments[2], permission)) {
+            api.sendMessage('Failed to set permissions. Please ensure permission has not already been ' + action + 'ed.', event.thread_id);
+        }
+        else {
+            api.sendMessage('Complete.', event.thread_id);
+        }
+    }
+    else if (action === 'grant' || action === 'revoke') {
+        if (!modify(event.arguments[2], event.thread_id, permission, action)) {
+            api.sendMessage('Failed to set permissions. Please ensure permission has not already been modified with ' + action + '.', event.thread_id);
+        }
+        else {
+            api.sendMessage('Complete.', event.thread_id);
+        }
+    }
+    else {
+        api.sendMessage('Only grant, revoke, create and delete are avalible.', event.thread_id);
+    }
+    
+    return false;
+};
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Newline required at end of file but not found.](https://www.codacy.com/app/theprojectshop/Kassy/file/2556616938/issues/source?bid=3282987&fileBranchId=3292072#l170)